tests/test_edgeql_functions.py

```diff
@@ -689,3 +689,176 @@ class TestEdgeQLFunctions(tb.QueryTestCase):
         ''', [
             [6.8],
         ])
+
+    @unittest.expectedFailure
+    async def test_edgeql_functions_str_01(self):
+        # at the very least the cast <str> should be equivalent to
+        # a call to str() without explicit format for simple scalars
+        await self.assert_query_result(r'''
+            WITH DT := current_datetime()
+            # FIXME: the cast has a "T" and the str doesn't for some reason
+            SELECT <str>DT = str(DT);
+
+            WITH D := current_date()
+            SELECT <str>D = str(D);
+
+            SELECT <str>123 = str(123);
+            SELECT <str>123.456 = str(123.456);
+            SELECT <str>123.456e-20 = str(123.456e-20);
+            # an unambiguous way to define a decimal is by casting
+            # from a string
+            SELECT <str><decimal>'123456789012345678901234567890.1234567890' =
+                str(<decimal>'123456789012345678901234567890.1234567890');
+        ''', [
+            [True],
+            [True],
+            [True],
+            [True],
+            [True],
+            [True],
+        ])
+
+    async def test_edgeql_functions_str_02(self):
+        await self.assert_query_result(r'''
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'YYYY-MM-DD');
+
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'YYYYBC');
+
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'FMDDth of FMMonth, YYYY');
+
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'CCth "century"');
+
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'Y,YYY Month DD Day');
+
+            # the format string doesn't have any special characters
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'foo');
+
+            # the format string doesn't have any special characters
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, '');
+        ''', [
+            {'2018-05-07'},
+            {'2018AD'},
+            {'7th of May, 2018'},
+            {'21st century'},
+            {'2,018 May       07 Monday   '},
+            {'foo'},
+            {},
+        ])
+
+    @unittest.expectedFailure
+    async def test_edgeql_functions_str_03(self):
+        await self.assert_query_result(r'''
+            WITH DT := <datetime>'2018-05-07 15:01:22.306916-05'
+            SELECT str(DT, 'HH:MI A.M.');
+        ''', [
+            # there's some implicit conversion happening, is that
```

Comment:
Postgres converts `timestamp with timezone` into the local timezone of the server.  We run tests with the UTC timezone.  Start the server with `--timezone=UTC` (I think we should do this by default).